.gitignore

```diff
@@ -0,0 +1,5 @@
+.idea/
+.vscode/
+.DS_Store
+.swp
+vendor/
```

Comment:
Could we reuse the ignore files in https://github.com/github/gitignore?

Reply:
Yes I think so. Do you mean this one specifically? https://github.com/github/gitignore/blob/master/Go.gitignore
Let me know if I miss anything else that I should be adding.

Reply:
Yeah, https://github.com/github/gitignore/blob/master/Global/Linux.gitignore and https://github.com/github/gitignore/blob/master/Global/macOS.gitignore also can be added

Reply:
Thanks. Added.